### ISSUE TITLE
Add serve options to polymer.json

### ISSUE THREAD

**Issue by TimvdLippe** [Polymer/polymer-cli#487]
Some projects use default configuration of serving. Right now a custom npm script is required with possibly a version of polyserve installed.

It would be nice if the args passed along to polyserve could be specified in a new field `serve` in the `polymer.json`. Then developers can just run `polymer serve` and it works without custom scripts.

**Comment by FredKSchott** [comment]
+1, there's a bigger discussion we need to have about where these options should live. `polymer.json` was always meant to describe your application, but it's now a place for some build options as well. Should all polymer-tooling-related options live there, or should we properly separate them out, into a separate `.polymerrc` CLI config file

**Comment by TimvdLippe** [comment]
Yes that is an interesting discussion I was wondering about too. Moreover, if you would want to extract these options for the various tools of the CLI, the config also needs to incorporate all arguments of all the tools. E.g. who takes care of the arguments, such that there is little duplication? Or just let the project-config import all used tools and just import their args, that's possible too

**Comment by Draccoz** [comment]
+1 from me. `polymer.json` is sort of `bower.json` or `package.json`. I don't think we should separate `.polymerrc` out of it, as build instructions are used in bower/npm json files as well.
Additionally, I would like to suggest sort of pre/post hooks (kinda npm scripts lifecycles, like `preinstall`, `postinstall` etc). What you guys think about that?

**Comment by justinfagnani** [comment]
Definitely agreed these options should go into a config. Whether they go in`polymer.json` is interesting too. In something like Bower, the line between config files is mostly "affects a dependent/installer" (bower.json) vs. "effects a contributor" (.bowerrc). npm just puts everything into package.json.

For us an appropriate line might be "settings for me only" vs "settings for my whole team". We could say that `serve` settings like hostname, aren't usually intended to be shared and should go in a separate config, but options like component-dir or package-name could be shared.

One option is to support all options in two files - one to be checked in and one not - and merge them. That would make the question of separate configs deferrable: just add all options to polymer.json then add support for a local config later.

**Comment by TimvdLippe** [comment]
> We could say that serve settings like hostname, aren't usually intended to be shared and should go in a separate config

In our case that is not true, as we are using a specific proxy url, protocol and serve location.

> One option is to support all options in two files - one to be checked in and one not - and merge them

Yes that sounds reasonable. The local file should have a higher priority than the "shared" file.

**Comment by justinfagnani** [comment]
Interesting. Good to have a data point that serving options child be shared. In that case I'd say we should move forward with all flags being settable in polymer.json, and add support for another config like polymer-local.json

**Comment by vdegenne** [comment]
I just wanted to comment as a lambda user. Because polymer 3.0 is using `"npm": true, "moduleResolution": "node"` in the `polymer.json` as arguments for the build process but also for the `polymer serve` tool, wouldn't it be nice to add `"proxy-path"` and `"proxy-target"` in the `polymer.json`. For me it was natural to set these options there, until I realized this is not working.
